dts: arm: stm32: add st,stm32-qdec nodes across H7/F7/G4/L4/L5/U5/G0/F1/F3/L0/L1

ritesh006 opened the pull request:
Adds st,stm32-qdec child nodes (disabled by default) under timers that
support encoder mode across these STM32 series:

- H7, F7: TIM1–TIM5, TIM8
- G4: TIM1–TIM4, TIM8
- L4: TIM1–TIM2
- L5: TIM1–TIM5, TIM8
- U5: TIM1–TIM5
- G0: TIM1, TIM3
- F1: TIM1–TIM4
- F3: TIM2
- L0: TIM2
- L1: TIM2–TIM4

DT coverage only; boards can opt in via overlays. No functional change.

Related-to: #88902

ritesh006 committed:
dts: arm: st: g4: add st,stm32-qdec child nodes (disabled)

Add st,stm32-qdec child nodes (disabled) under TIM nodes in STM32G4
DTSI files. Nodes are disabled by default; boards may enable them via
overlays. No functional change.

Signed-off-by: Ritesh Kudkelwar <ritesh.kumar0793@gmail.com>

ritesh006 committed:
dts: arm: st: h7: add st,stm32-qdec child nodes (disabled)

Add st,stm32-qdec child nodes (disabled) under TIM nodes in STM32H7
DTSI files. Nodes are disabled by default; boards may enable them via
overlays. No functional change.

Signed-off-by: Ritesh Kudkelwar <ritesh.kumar0793@gmail.com>

ritesh006 committed:
dts: arm: st: f7: add st,stm32-qdec child nodes (disabled)

Add st,stm32-qdec child nodes (disabled) under TIM nodes in STM32F7
DTSI files. Nodes are disabled by default; boards may enable them via
overlays. No functional change.

Signed-off-by: Ritesh Kudkelwar <ritesh.kumar0793@gmail.com>

ritesh006 committed:
dts: arm: st: l4: add st,stm32-qdec child nodes (disabled)

Add st,stm32-qdec child nodes (disabled) under TIM nodes in STM32L4
DTSI files. Boards can enable them in overlays. No functional change.

Signed-off-by: Ritesh Kudkelwar <ritesh.kumar0793@gmail.com>

ritesh006 committed:
dts: arm: st: g0: add st,stm32-qdec child nodes (disabled)

Add st,stm32-qdec child nodes (disabled) under TIM nodes in STM32G0
DTSI files. Boards can enable them in overlays. No functional change.

Signed-off-by: Ritesh Kudkelwar <ritesh.kumar0793@gmail.com>

erwango commented:
Thanks @ritesh006, have you been able to test this ?
Also, at the minimum we'd need a way to test those in CI, for instance by adding overlays here:samples/sensor/qdec/boards

gautierg-st reviewed (commented):
Since we're looking at full completion of the various dtsi, it seems you've missed a few timers across different families.
Please also split your last commit to do one commit per family.

etienne-lms reviewed (commented):
See also the "compliance checks" reports: empty lines in-between nodes should emptied from space characters.

ritesh006 commented on dts/arm/st/g0/stm32g0.dtsi:
done

ritesh006 commented on dts/arm/st/f3/stm32f3.dtsi:
stm32f302.dtsi for this timer4 not available in code 
stm32f3.dtsi for this timer3 not available in code
stm32f303.dtsi for this timer4, 8, 20 not available in code

gautierg-st replied:
Oops, fixed now. Thanks!

etienne-lms replied:
AFAICT, the requested updates have been addressed.

ritesh006 commented on dts/arm/st/f1/stm32f1.dtsi:
done

ritesh006 commented on dts/arm/st/f1/stm32f1.dtsi:
```diff
@@ -284,6 +284,12 @@
+			qdec {
+				compatible = "st,stm32-qdec";
+				status = "disabled";
+				st,input-filter-level = <0>;
```
done

etienne-lms reviewed (commented):
Could you squash changes in commit *"dts: added timers as instructed and"* into their relative commits in your series.
By the way, I don't think `Ref: ` is a valid commit message tag. Please replace with a `Fixes: ` tag:
```patch
-Ref: #88902
+Fixes: #88902
```

ritesh006 committed:
dts: arm: st: f2: clean QDEC node properties

Reorder QDEC node properties for consistency in STM32F2 DTSI files.
No functional change.

Signed-off-by: Ritesh Kudkelwar <ritesh.kumar0793@gmail.com>

etienne-lms reviewed (commented):
Could you fix the commit message (replace `Ref:` with `Fixes:`) in the 5 first commits of your series?

Commits *"dts: arm: stm32: add qdec to TIM1–TIM5 & TIM8 on STM32F2"*,
*"dts: arm: stm32: add qdec nodes on STM32F4 base dtsi"*,
*"dts: arm: stm32: add qdec nodes on STM32F405"* and
*"dts: arm: stm32: add qdec nodes on STM32F412"*
are lacking a commit message body and a `Signed-off-by:`  tag. Moverover, the right description should rather be "cleanup QDEC nodes properties", without the `Fixes:` tag reference since these 2 commits do not fix the related issues. Maybe the 3 latter could be squashed into a single commit *"dts: arm: st: f4: clean QDEC node properties"*.

Otherwise, LGTM.

etienne-lms reviewed (commented):
Could you update the commit message of commit "dts: arm: stm32: add qdec to TIM1–TIM5 & TIM8 on STM32F2"?

```patch
 dts: arm: stm32: add qdec to TIM1–TIM5 & TIM8 on STM32F2
 
-Add st,stm32-qdec child nodes (disabled) under TIM1–TIM5 and TIM8 for
-STM32F2. DT coverage only so boards can opt in via overlays. No functional
-change.
+Place 'status' last, remove trailing whitespace on blank lines, and
+align QDEC node properties across F2 base DTSI filee. No functional change.
 
-Fixes: https://github.com/zephyrproject-rtos/zephyr/issues/88902
 Signed-off-by: ...
```

Also, commit *"dts: arm: st: f4: clean QDEC node properties"* includes fixup changes in *dts/arm/st/l5/stm32l5.dtsi*. These should be squashed into commit *"dts: arm: stm32: add qdec across G0/L5/U5/F1/F3/L0/L1"*.

ritesh006 committed:
dts: arm: st: f4: clean QDEC node properties

Reorder and tidy QDEC node properties in STM32F4 DTSI files.
No functional change.

Signed-off-by: Ritesh Kudkelwar <ritesh.kumar0793@gmail.com>

ritesh006 commented:
ready for review

etienne-lms reviewed (commented):
You need to add a commit message body and a Signed-off-by tag to commit
*"dts: arm: st: f4: clean QDEC node properties"*.

ritesh006 commented:
Ready for review

etienne-lms commented on dts/arm/st/u5/stm32u5.dtsi:
```diff
@@ -21,12 +21,13 @@
 		zephyr,entropy = &rng;
 		zephyr,flash-controller = &flash;
-	};
+};
```
Typo, please discard this change.

etienne-lms commented on dts/arm/st/u5/stm32u5.dtsi:
```diff
@@ -21,12 +21,13 @@
 #include <zephyr/dt-bindings/adc/stm32u5_adc.h>
 #include <zephyr/dt-bindings/power/stm32_pwr.h>
 #include <freq.h>
+#include <zephyr/dt-bindings/sensor/qdec_stm32.h>
```
nitpicking: could you swap the 2 lines above, to group the zephyr headers?

etienne-lms reviewed (commented):
2 last minor comments to address. Otherwise all commits look all good to me.
Thanks for the work.

etienne-lms reviewed (commented):
Sorry, I missed 2 more comments, related to commit message to not mention changes not applied:

For commit "dts: arm: stm32: add qdec to TIM1–TIM5 & TIM8 on STM32F2":
```patch
-dts: arm: stm32: add qdec to TIM1–TIM5 & TIM8 on STM32F2
+dts: arm: st: f2: clean QDEC node properties
 
-Place 'status' last, remove trailing whitespace on blank lines, and
-align QDEC node properties across F2 base DTSI file. No functional change.
+Place 'status' last in QDEC node properties in F2 base DTSI file for
+consistency. No functional change.
```

For commit "dts: arm: st: f4: clean QDEC node properties":
```patch
-Place 'status' last, remove blank lines, and fix indentation
-in QDEC nodes across F4 base and variants. No functional change.
+Place 'status' last in QDEC nodes and include related binding header
+file across F4 base and variants for consistency. No functional change.

```

ritesh006 commented:
Ready for review. Thank you!

etienne-lms reviewed (commented):
qdec nodes in stm32u5.dtsi seem to have disappeared. Only the addition of the header file inclusion remains.

ritesh006 commented:
ready for review

ritesh006 commented on dts/arm/st/u5/stm32u5.dtsi:
```diff
@@ -661,6 +686,12 @@
 				status = "disabled";
 			};
+
+			qdec {
```
Hi I am not seeing this changes if you compare latest commit there only necessary changes available please reffer this screenshot  link:- https://github.com/zephyrproject-rtos/zephyr/compare/9c23b37d0c85fede5194f41868fd7be7ff9a1091..166dfd81b6d5c88201678704d13fe7fdd45c0012
<img width="534" height="716" alt="image" src="https://github.com/user-attachments/assets/36501503-1696-46fd-b01f-23deb8806305" />

ritesh006 committed:
dts: arm: st: f1: add st,stm32-qdec child nodes (disabled)

Add st,stm32-qdec child nodes (disabled) under TIM nodes in the
STM32F1 DTSI files. Boards can enable encoder/decoder functionality
using overlays. No functional change.

Signed-off-by: Ritesh Kudkelwar <ritesh.kumar0793@gmail.com>

ritesh006 committed:
dts: arm: st: f3: add st,stm32-qdec child node (disabled)

Add st,stm32-qdec child node (disabled) under TIM nodes in STM32F3
DTSI files. Boards can enable the node via overlays. No functional change.

Signed-off-by: Ritesh Kudkelwar <ritesh.kumar0793@gmail.com>

ritesh006 committed:
dts: arm: st: l0: add st,stm32-qdec child node (disabled)

Add st,stm32-qdec child node (disabled) under TIM nodes in STM32L0
DTSI files. Boards can enable it in overlays. No functional change.

Signed-off-by: Ritesh Kudkelwar <ritesh.kumar0793@gmail.com>

ritesh006 committed:
dts: arm: st: l1: add st,stm32-qdec child nodes (disabled)

Add st,stm32-qdec child nodes (disabled) under TIM nodes in STM32L1
DTSI files. Boards can enable them in overlays. No functional change.

Signed-off-by: Ritesh Kudkelwar <ritesh.kumar0793@gmail.com>

ritesh006 committed:
dts: arm: st: l5: add st,stm32-qdec child nodes (disabled)

Add st,stm32-qdec child nodes (disabled) under TIM nodes in STM32L5
DTSI files. Nodes are disabled by default so boards can enable encoder
mode via overlays. No functional change.

Signed-off-by: Ritesh Kudkelwar <ritesh.kumar0793@gmail.com>

ritesh006 committed:
dts: arm: st: u5: add st,stm32-qdec child nodes (disabled)

Add st,stm32-qdec child nodes (disabled) under TIM nodes in STM32U5
DTSI files. Nodes are disabled by default; boards can enable encoder
mode via overlays. No functional change.

Signed-off-by: Ritesh Kudkelwar <ritesh.kumar0793@gmail.com>

gautierg-st reviewed (commented):
Almost there, only a few missing nodes and one to remove

ritesh006 commented:
please review

ritesh006 commented:
please someone review this

gautierg-st reviewed (commented):
LGTM for the dtsi part.
There remains the comment from @erwango regarding the tests https://github.com/zephyrproject-rtos/zephyr/pull/95624#issuecomment-3264823255

etienne-lms reviewed (commented):
1 nitpicking comment on the commit messages (header lines and bodies). Some commit messages to not reflect the changes since the list only a subset of the TIMx instances modified. I've listed below the faulty commits and the TIM nodes actually modified.

*"dts: arm: st: f1: add qdec to TIM1–TIM4 (disabled)"*   => TIM1/2/3/4/5
*"dts: arm: st: f3: add qdec to TIM2 (disabled)"* => TIM12/3/4/5/8/20
*"dts: arm: st: g0: add qdec to TIM1 & TIM3 (disabled)"* =>  TIM1/2/3/4
*"dts: arm: st: g4: add qdec to TIM1–TIM4 & TIM8 (disabled)"* => TIM1/2/3/4/5/8/20
*"dts: arm: st: h7: add qdec to TIM1–TIM5 & TIM8 (disabled)"* => TIM1/2/3/4/5/8/23/24
*"dts: arm: st: l0: add qdec to TIM2 (disabled)"* => TIM2/3
*"dts: arm: st: f1: add qdec to TIM1–TIM4 (disabled)"*   => TIM1/2/3/4/5
*"dts: arm: st: f3: add qdec to TIM2 (disabled)"* => TIM1/2/3/4/5/8/20
*"dts: arm: st: g0: add qdec to TIM1 & TIM3 (disabled)"* =>  TIM1/2/3/
*"dts: arm: st: g4: add qdec to TIM1–TIM4 & TIM8 (disabled)"* => TIM1/2/3/4/5/8/20
*"dts: arm: st: h7: add qdec to TIM1–TIM5 & TIM8 (disabled)"* => TIM1/2/3/4/5/8/23/24
"dts: arm: st: l0: add qdec to TIM2 (disabled)"* => TIM2/3/4/5
"dts: arm: st: l1: add qdec to TIM2–TIM4 (disabled)"* => TIM2/3/4/5
"dts: arm: st: l4: add qdec to TIM1 & TIM2 (disabled)"* => TIM1/2/3/4/5/8

Here is a proposal of template for the message header line & body (based on your wording), base for example on G4 case (this is only a proposal, feel free to rephrase):
```
dts: arm: st: g4: add qdec (disabled) to applicable TIM nodes

Add st,stm32-qdec child nodes under TIM1/2/3/4/5/8/20 nodes
on STM32G4 SoCs DTSI file when the SoC sub-series embeds
the related TIM instance.

Nodes are disabled by default so boards can enable encoder mode via
overlays. No functional change.

Signed-off-by: ...
```

ritesh006 commented:
Hi @etienne-lms due family loss I went to hometown as soon as I reach to my place I would like to work start again. Thankyou for understanding

gautierg-st commented:
Hi @ritesh006,
I'm very sorry for your loss. Take all the time you need, there is no rush.

erwango commented:
Non blocking comment on my side:
You can omit "(disabled)" mention. Nodes in .dtsi are all disabled (aside from few specific exceptions), this is the general concept.

ritesh006 commented:
Ready For Review

ritesh006 commented:
> disabled

Ready for Review

etienne-lms reviewed (commented):
I suggest you remove the phrasing like:
> This commit actually modifies qdec nodes under: TIM1
(see files modified in this commit:
  dts/arm/st/f1/stm32f1.dtsi
  dts/arm/st/f1/stm32f103Xc.dtsi
  dts/arm/st/f1/stm32f105.dtsi
)

from the all commit messages. They don't really help and the mentioned TIMs IDs are not always accurate.

For commit *"dts: arm: st: f2: clean QDEC node properties"*
and commit *"dts: arm: st: f4: clean QDEC node properties"*:
The commit message mentioned redundant/incorrect properties. This is not what these commit do. No need to say that the commit do not add new nodes. Consider that commit messages, in a generic way, are there to help someone in few months or years from now to look at individual commits and understand what is done and why it's done. Prefer to simply says what it done:
> Cleanup QDEC node properties ordering in STM32Fx SoC DTSI files.
No functional change.

Otherwise, content of the changes still looks all good to me.

etienne-lms reviewed (commented):
Changes look all good to me.
My sole concern are the commit message: they list the modified TIM instance but for some commits there are TIM instances not mentioned in the related commit message.

See below the list of the commit message to update:

```
dts: arm: st: f1: add qdec to applicable TIM's

Add st,stm32-qdec child nodes (disabled) under TIM1/2/3/4/5 on STM32F1.
Boards can enable them in overlays. No functional change.

Signed-off-by: ...
```
```
dts: arm: st: f3: add qdec to applicable TIM's
    
Add st,stm32-qdec child node (disabled) under TIM1/2/3/4/5/8/20 on STM32F3.
...
```
```
dts: arm: st: g0: add qdec to applicable TIM's

Add st,stm32-qdec child nodes (disabled) under TIM1/2/3/4 on STM32G0.
...
```
```
dts: arm: st: g4: add qdec to applicable TIM's

Add st,stm32-qdec child nodes under TIM1/2/3/4/5/8/20 on STM32G4.
...
```
```
dts: arm: st: h7: add qdec to applicable TIM's

Add st,stm32-qdec child nodes under TIM1/2/3/4/5/8/23/24 on STM32H7.
...
```
```
dts: arm: st: l0: add qdec to applicable TIM's

Add st,stm32-qdec child node (disabled) under TIM2 and TIM3 on STM32L0.
...
```
```
dts: arm: st: l1: add qdec to applicable TIM's

Add st,stm32-qdec child nodes (disabled) under TIM2/3/4/5 on STM32L1.
...
```
```
dts: arm: st: l4: add qdec to applicable TIM's

Add st,stm32-qdec child nodes (disabled) under TIM1/2/3/4/5/8 on STM32L4.
...
```

etienne-lms reviewed (approved):
All good to me. Thanks.
